src/highlight.rs

```diff
@@ -232,6 +248,88 @@ fn is_close_bracket(bracket: u8) -> bool {
     memchr(bracket, CLOSES).is_some()
 }
 
+pub(crate) fn split_highlight<'a>(src: &'a str, offset: usize)
```

Comment:
By the way, my other PR, improves the function to use offset in the highlighted source instead: https://github.com/kkawakam/rustyline/pull/378/files#diff-9df84bc451a5211110d67dd9b47c627ded197d37c6e04e38b5a8d0fe098257d3R252
(I can backport it here, if this will help).
 
Comparing with the original is probably sub par because it's harder to optimize. I.e. with current semantics we can use `memchr` to skip until first `\x1b` char. While this PR doesn't do that, I should be improved later.

Also we [replace characters by highlighter for continuation prompt](https://github.com/kkawakam/rustyline/pull/372/files#diff-869adc680365c930a66ac6afcc025192ef540f78e5de8578f46be6c48e326b6dR43), so we need to create another mechanism for that.

Reply:
> By the way, my other PR, improves the function to use offset in the highlighted source instead

Hm, that's not quite what I mean either, I'll get back to you later today on that though, I got sidetracked yesterday.

> Comparing with the original is probably sub par because it's harder to optimize

These lines tend to be small enough that the performance here isn't visible, and the slowest thing by far is the IO anyway.

I'm way more concerned about correctness than performance here, mostly because the bugs in this tend to be pretty annoying, both as a user of the library, and as a user of the end application...

> Also we replace characters by highlighter for continuation prompt, so we need to create another mechanism for that.

Yeah, that's what I meant when i said: "and you can then see if it looks like a SGR sequence for the prompt support for the splitting". (That said, I also think we shouldn't bother with that in cases where the continuation prompt is either empty or all whitespace, but that's neither here nor there, and is relatively easy to implement no matter what).

Reply:
Another use case for different chars in highlighted and non-highlighted text if someone wants to highlight some whitespace. You can do that with just background color, but it takes too much attention and to hard to remember which colors mean what. I.e. in my vim config have this:
```
set list listchars=tab:·⁖,trail:¶,nbsp:×
```
Which means tabs are replaced by `·⁖`, trailing characters are replaced by `¶` and non-breaking spaces are replaced by `×`. This shows characters in more subtle and more recognizable way. So I can see these things in the code, but they are saved as original characters.

While I haven't implemented that in any rustyline use case yet, I can imagine it to be useful. And this has to be done in highlighter but not in original text.